sdk/go/common/resource/plugin/context.go

```diff
@@ -50,13 +51,22 @@ func NewContext(d, statusD diag.Sink, host Host, cfg ConfigSource,
 	parentSpan opentracing.Span) (*Context, error) {
 
 	root := ""
-	return NewContextWithRoot(d, statusD, host, cfg, pwd, root, runtimeOptions, disableProviderPreview, parentSpan)
+	projPath, err := workspace.DetectProjectPath()
+	var project *workspace.Project
+	if err == nil && projPath != "" {
+		project, err = workspace.LoadProject(projPath)
+		if err != nil {
+			project = nil
+		}
+	}
+	return NewContextWithRoot(d, statusD, host, cfg, pwd, root, runtimeOptions,
+		disableProviderPreview, parentSpan, project)
 }
 
-// Variation of NewContext that also sets known project Root.
+// Variation of NewContext that also sets known project Root. Additionally accepts Project
 func NewContextWithRoot(d, statusD diag.Sink, host Host, cfg ConfigSource,
 	pwd, root string, runtimeOptions map[string]interface{}, disableProviderPreview bool,
-	parentSpan opentracing.Span) (*Context, error) {
+	parentSpan opentracing.Span, project *workspace.Project) (*Context, error) {
```

Comment:
```suggestion
	parentSpan opentracing.Span, projectPlugins []*workspace.PluginInfo) (*Context, error) {
```

Given otherwise you have to deal with "project" being nil below anyway.